components/cors-mgt/org.wso2.carbon.identity.cors.mgt.core/src/main/java/org/wso2/carbon/identity/cors/mgt/core/internal/impl/CORSManagementServiceImpl.java

```diff
@@ -0,0 +1,333 @@
+import org.wso2.carbon.identity.cors.mgt.core.exception.CORSManagementServiceClientException;
+import org.wso2.carbon.identity.cors.mgt.core.exception.CORSManagementServiceException;
+import org.wso2.carbon.identity.cors.mgt.core.exception.CORSManagementServiceServerException;
+import org.wso2.carbon.identity.cors.mgt.core.internal.CORSManagementServiceHolder;
+import org.wso2.carbon.identity.cors.mgt.core.internal.function.CORSOriginToAttribute;
+import org.wso2.carbon.identity.cors.mgt.core.internal.function.CORSOriginToResourceAdd;
+import org.wso2.carbon.identity.cors.mgt.core.internal.function.ResourceToCORSOrigin;
+import org.wso2.carbon.identity.cors.mgt.core.model.CORSOrigin;
+
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.wso2.carbon.identity.configuration.mgt.core.constant.ConfigurationConstants.ErrorMessages.ERROR_CODE_RESOURCE_TYPE_DOES_NOT_EXISTS;
+import static org.wso2.carbon.identity.cors.mgt.core.constant.ErrorMessages.ERROR_CODE_CORS_CRUD;
+import static org.wso2.carbon.identity.cors.mgt.core.constant.ErrorMessages.ERROR_CODE_EMPTY_LIST;
+import static org.wso2.carbon.identity.cors.mgt.core.constant.ErrorMessages.ERROR_CODE_INVALID_ORIGIN;
+import static org.wso2.carbon.identity.cors.mgt.core.constant.ErrorMessages.ERROR_CODE_ORIGIN_NOT_PRESENT;
+import static org.wso2.carbon.identity.cors.mgt.core.constant.ErrorMessages.ERROR_CODE_ORIGIN_PRESENT;
+import static org.wso2.carbon.identity.cors.mgt.core.internal.Constants.CORS_ORIGIN_RESOURCE_NAME;
+import static org.wso2.carbon.identity.cors.mgt.core.internal.Constants.CORS_ORIGIN_RESOURCE_TYPE_DESCRIPTION;
+import static org.wso2.carbon.identity.cors.mgt.core.internal.Constants.CORS_ORIGIN_RESOURCE_TYPE_NAME;
+
+/**
+ * Implementation of the CORSService.
+ */
+public class CORSManagementServiceImpl implements CORSManagementService {
+
+    private static final Log log = LogFactory.getLog(CORSManagementServiceImpl.class);
+
+    @Override
+    public List<CORSOrigin> getCORSOrigins(String tenantDomain) throws CORSManagementServiceException {
+
+        validateTenantDomain(tenantDomain);
+        try {
+            FrameworkUtils.startTenantFlow(tenantDomain);
+            addCORSOriginResourceTypeIfNotExists();
+
+            Resource resource = getConfigurationManager().getResource(CORS_ORIGIN_RESOURCE_TYPE_NAME,
+                    CORS_ORIGIN_RESOURCE_NAME);
+            List<CORSOrigin> corsOrigins;
+            if (resource == null) {
+                corsOrigins = new ArrayList<>();
+            } else {
+                corsOrigins = new ResourceToCORSOrigin().apply(resource);
+            }
+
+            return corsOrigins;
+        } catch (ConfigurationManagementException | IOException e) {
+            throw handleServerException(ERROR_CODE_CORS_CRUD, e, tenantDomain, "retrieving");
```

Comment:
It's not good to pass text information that should be in the message as arguments. It breaks readability.